includes/wikia/services/WikiService.class.php

```diff
@@ -97,8 +101,47 @@ function() use ( $wiki, $useMaster, $excludeBots, $limit ) {
 		return $userIds;
 	}
 
-	private static function getAdminIdsFromDB( DatabaseBase $db, $excludeBots = false, $limit = null ) {
-		$conditions = [ 'ug_group' => self::ADMIN_GROUPS ];
+	/**
+	 * get list of wiki chatmoderator ids
+	 *
+	 * @param integer $wikiId - wiki Id (default: current wiki Id)
+	 * @param bool    $useMaster - flag that describes if we should use masted DB (default: false)
+	 * @param bool    $excludeBots - flag that describes if bots should be excluded from moderatorlist (default: false)
+	 * @param integer $limit - limit for the number of moderators
+	 *
+	 * @return array of $userIds
+	 */
+	public function getWikiChatmoderatorIds( $wikiId = 0, $useMaster = false, $excludeBots = false, $limit = null ) {
+		$wikiId = empty( $wikiId ) ? $this->wg->CityId : $wikiId ;
+		$wiki = WikiFactory::getWikiById( $wikiId );
+
+		if ( empty( $wiki ) || $wiki->city_public != 1 ) {
+			return [];
+		}
+
+		$userIds = [];
+
+		// Get moderators
+		$memKey = wfSharedMemcKey( 'wiki_chatmoderator_ids', $wikiId, $excludeBots, $limit );
+		$moderatorIds = WikiaDataAccess::cache(
+			$memKey,
+			self::WIKI_ADMIN_IDS_CACHE_TTL,
+			function() use ( $wiki, $useMaster, $excludeBots, $limit ) {
+				$dbName = $wiki->city_dbname;
+				$dbType = $useMaster ? DB_MASTER : DB_SLAVE;
+				$db = wfGetDB( $dbType, [], $dbName );
+
+				return self::getUserIdsFromDB( $db, $excludeBots, $limit, self::MODERATOR_GROUPS );
+			}
+		);
+
+		$userIds = array_unique( array_merge( $userIds, $moderatorIds ) );
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$userIds". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1488)